system/Database/BaseConnection.php

```diff
@@ -897,7 +897,7 @@ public function prepare(Closure $func, array $options = [])
 
         $this->pretend(false);
 
-        if ($sql instanceof QueryInterface) {
+        if ($sql instanceof Query) {
```

Comment:
Again - BC break since they can currently pass custom implementations.

Reply:
`QueryInterface` does not have `getOriginalQuery()`.
So it can't be used.

Reply:
@lonnieezell

Reply:
Would it be less breaking to add `getOriginalQuery` to the interface? I realize they can just extend the original Query class to do custom implementations but that defeats the whole purpose of having an interface to begin with. 

Without looking at the history of things I want to say that method came after the initial release, but even if it didn't I think the bug would be that it wasn't included.